src/Services/Catalog/Command/Domain/ValueObjects/Money.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace Domain.ValueObjects;
+
+public readonly record struct Money(Currency Currency, decimal Value)
+{
+    public static Money operator +(Money money, Money other)
+        => money with { Value = money.Value + other.Value };
+
+    public static Money operator -(Money money, Money other)
+        => money with { Value = money.Value - other.Value };
+
+    public static Money operator *(Money money, Money other)
+        => money with { Value = money.Value * other.Value };
+
+    public static Money operator *(Money money, int integer)
+        => money with { Value = money.Value * integer };
+
+    public static Money operator /(Money money, Money other)
+    {
+        if (other.Value is 0) throw new DivideByZeroException();
+        return money with { Value = money.Value / other.Value };
+    }
+
+    public static Money operator %(Money money, Money other)
+    {
+        if (other.Value is 0) throw new DivideByZeroException();
+        return money with { Value = money.Value % other.Value };
+    }
+
+    public static bool operator >(Money money, Money other)
+        => money.Value > other.Value;
+
+    public static bool operator <(Money money, Money other)
+        => money.Value < other.Value;
+
+    public static implicit operator string(Money money)
+        => $"{money.Value} {money.Currency.Symbol}";
```

Comment:
this symbol believe is first then value.

See example below

![image](https://user-images.githubusercontent.com/17841549/213246964-fd1f0cff-d0b4-4618-be54-70126cc2bc1d.png)